### ISSUE TITLE
Arrays referencing multiple schemas with oneOf are resolved as type unknown. 

### ISSUE THREAD

**Issue by afohlmeister-smatch** [drwpow/openapi-typescript#1148]
**Description**

When referencing multiple schemas within an array, the resulting typescript type is 

*Input*

```
{
    "global.markets": {
        "items": {
            "$ref": "#/$defs/global.market"
        },
        "uniqueItems": true
    },
    "global.market": {
        "oneOf": [
            {
                "$ref": "#/$defs/global.region-code",
                "title": "Region"
            },
            {
                "$ref": "#/$defs/global.country-code",
                "title": "Country"
            }
        ]
    },
    "global.region-code": {
        "enum": [
            "northern_africa",
            "eastern_africa",
            "oceania"
        ]
    },
    "global.country-code": {
        "enum": [
            "ABW",
            "AFG",
            "AGO"
        ]
    }
}
```
*Result:*
```
 "global.market": components["schemas"]["global.region-code"] | components["schemas"]["global.country-code"];
 "global.markets": unknown;
````

| Name                 | Version                      |
| :------------------- | :--------------------------- |
| `openapi-typescript` | `6.2.6`                      |
| Node.js              | `v19.6.0`                      |
| OS + version         | `macOS 13.0.1` |

**Expected result**

`global.markets` should be types as an array referencing `global.market`

**Checklist**

- [x] My OpenAPI schema passes the [Redocly validator](https://redocly.com/docs/cli/commands/lint/) (`npx @redocly/cli@latest lint`)
- [ ] I’m willing to open a PR (see [CONTRIBUTING.md](https://github.com/drwpow/openapi-typescript/blob/main/CONTRIBUTING.md))

**Comment by afohlmeister-smatch** [comment]
Hey there, because I'm idiot, I checked your issues regarding tuples, arrays and items and completely missed https://github.com/drwpow/openapi-typescript/issues/1131 - this is probably a related issue.

**Comment by drwpow** [comment]
The latest release fixes #1131, but I’ll leave this issue open to double-check; this may be a different bug.

**Comment by drwpow** [comment]
Going to mark the issue as closed, because I believe the issue is resolved? But please let me know if updated versions didn’t address the issue.
